src/FileConventions/Library.fs

```diff
@@ -104,3 +103,63 @@ let EolAtEof(fileInfo: FileInfo) =
                 False
         else
             True
+
+let DetectInconsistentVersionsInGitHubCIWorkflow(fileInfos: seq<FileInfo>) =
+    fileInfos
+    |> Seq.iter(fun fileInfo -> assert (fileInfo.FullName.EndsWith ".yml"))
+
+    let inconsistentPulumiVersions =
+        fileInfos
+        |> Seq.map(fun fileInfo -> File.ReadLines fileInfo.FullName)
+        |> Seq.map(fun fileLines ->
+            fileLines
+            |> Seq.filter(fun line -> line.Contains "pulumi-version:")
+            |> Seq.map(fun line -> line.Substring(line.IndexOf(":") + 1))
+            |> Seq.map(fun line -> line.Trim())
+        )
+        |> Seq.concat
+        |> Set.ofSeq
+        |> Seq.length
+        |> (fun length -> length > 1)
+
+    let versionRegex =
+        Regex("\\suses: ([^\\s]*)@v([^\\s]*)\\s", RegexOptions.Compiled)
```

Comment:
@realmarv after `uses:` shouldn't it try to match with any whitespace? (e.g. `s*`)

Reply:
are npm packages in .ymls? no; plus, we're covering only GitHubCI for now in this functionality; if we want to augment the functionality later, we would need new tests first; don't deviate from TDD please

Reply:
fair enough